File: solutions/q345/q345_solution.kt
import java.util.Collections

class Solution {
    fun reverseVowels(s: String): String {
        val vowels = setOf('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O','U')
        val chars = s.toMutableList()

        var i = 0
        var j = chars.size - 1

        while( i < j ) {
            if(chars[i] !in vowels)
                i++
            else if(chars[j] !in vowels)
                j--
            else{
                Collections.swap(chars,i,j)
                i++
                j--
            }
        }

        return chars.joinToString("")
    }
}

File: solutions/q345/q345_solution.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        s = list(s)
        vowels = set('aeiouAEIOU')

        i = 0
        j = len(s) - 1

        while(i < j):
            if(s[i] not in vowels):
                i += 1
            elif(s[j] not in vowels):
                j -= 1
            else:
                s[i], s[j] = s[j], s[i]
                i += 1
                j -= 1
                
        return ''.join(s)
